### ISSUE TITLE
Supported instance type being rejected in region eu-north-1

### ISSUE THREAD

**Issue by sonofachamp** [aws/amazon-ecs-cli#1006]
A user reached out with the following issue using `ecs-cli version 1.18.1 (7e9df84)`:

`$ ecs-cli up  --capability-iam --size 4 --instance-type c4.large --launch-type EC2`

```
WARN[0000] You will not be able to SSH into your EC2 instances without a key pair. 
INFO[0001] Using recommended Amazon Linux 2 AMI with ECS Agent 1.39.0 and Docker version 18.09.9-ce 
INFO[0002] Created cluster                               cluster=test region=eu-north-1
FATA[0003] Error executing 'up': ValidationError: Parameter EcsInstanceType failed to satisfy constraint: must be a valid EC2 instance type.
	status code: 400, request id: <redacted>
```

`$ ecs-cli up  --capability-iam --size 4 --instance-type t3.medium --launch-type EC2`

```
WARN[0000] You will not be able to SSH into your EC2 instances without a key pair. 
INFO[0001] Using recommended Amazon Linux 2 AMI with ECS Agent 1.39.0 and Docker version 18.09.9-ce 
INFO[0002] Created cluster                               cluster=test region=eu-north-1
FATA[0003] Error executing 'up': ValidationError: Parameter EcsInstanceType failed to satisfy constraint: must be a valid EC2 instance type.
	status code: 400, request id: <redacted>
```

I validated that `t3.medium` and `c4.large` are indeed available in `eu-north-1`:
`AWS_REGION=eu-north-1 aws ec2 describe-instance-types --query InstanceTypes[].InstanceType`

I asked them to try cli version 1.18.0 due to the changes introduce in 1.18.1 that affected how we validate instance types: https://github.com/aws/amazon-ecs-cli/pull/958

It works in version 1.18.0:
`$ ecs-cli up  --capability-iam --size 4 --instance-type t3.medium --launch-type EC2`
```
WARN[0000] You will not be able to SSH into your EC2 instances without a key pair. 
INFO[0001] Using recommended Amazon Linux 2 AMI with ECS Agent 1.39.0 and Docker version 18.09.9-ce 
INFO[0001] Created cluster                               cluster=test region=eu-north-1
INFO[0002] Waiting for your cluster resources to be created... 
INFO[0002] Cloudformation stack status                   stackStatus=CREATE_IN_PROGRESS
INFO[0064] Cloudformation stack status                   stackStatus=CREATE_IN_PROGRESS
```

This seems to be a potential regression from version 1.18.0 -> 1.18.1 with how we're validating input instance types.

**NOTE:** The scope of affected regions and instance types is currently unknown.

**Comment by sonofachamp** [comment]
The underlying issue here is that the `EcsInstanceType` CloudFormation template parameter is validated by CloudFormation against the `AllowedValues` field. The `DefaultECSInstanceType` `t2.micro` is not valid in region `eu-north-1`.
https://github.com/aws/amazon-ecs-cli/blob/e370a691408cb2a6948e76f9259206c23fba7c51/ecs-cli/modules/clients/aws/cloudformation/cluster_template.go#L95

I determined this by debugging the cli with this command:
`ecs-cli up --instance-type t3.medium --region eu-north-1`

The call to `DescribeInstanceTypeOfferings` was indeed returning `t3.medium` and `c4.large` as supported instance types, but it was the `Default` value in the template being rejected.

https://github.com/aws/amazon-ecs-cli/blob/e370a691408cb2a6948e76f9259206c23fba7c51/ecs-cli/modules/clients/aws/cloudformation/cluster_template.go#L115-L120

I confirmed this by modifying the `DefaultECSInstanceType` to be `t3.medium` and the command ran just fine.

Potential paths forward:
* Don't include the Default value in the template if a custom instance type is specified
* Ensure the `DefaultECSInstanceType` is a value that is available in **all** regions
* Ensure the `DefaultECSInstanceType` is one of the values returned by the call to `DescribeInstanceTypeOfferings`
* Remove the default value (probably a breaking change so doubt this is the right choice)
